documentation/docs/utils/LoginManager.ts

```diff
@@ -1,50 +1,85 @@
-    this.token = this.handleLoginFromRedirect();
+  private readonly _clientId = '2fb875d7-80b5-414a-9983-8f053372d760';
+  private readonly _tenantId = '48d5043c-cf70-4c49-881c-c638f5796997';
+  
+  private readonly _userScopes = ['User.Read'];
+  private readonly _cdfScopes = ['https://api.cognitedata.com/user_impersonation'];
+  
+  private readonly _msalInstance: PublicClientApplication;
 
-    this.isLoggedIn = !!this.token;
-    this.project = project;
-    this.cluster = cluster;
+  public get isLoggedIn() {
+    return this._isLoggedIn;
   }
 
-  private handleLoginFromRedirect() {
-    const tokens = extractTokensFromUrl();
-    return tokens?.accessToken;
+  constructor() {
+    this._isLoggedIn = false;
+
+    this._msalInstance = this.setupMsalInstance();
   }
 
-  async getToken(): Promise<string> {
-    if (!this.token) {
-      await this.loginWithRedirect();
+  public async getToken(): Promise<string> {
+    const account = this._msalInstance.getActiveAccount();
+
+    if (!account) {
+      throw Error("No local account found");
     }
 
-    return this.token!;
+    const { accessToken } = await this._msalInstance.acquireTokenSilent({
+      account,
+      scopes: this._cdfScopes
+    });
+
+    return accessToken;
+  }
+
+  public async loginPopup(): Promise<void> {
+    await this._msalInstance.loginPopup({ scopes: this._userScopes, extraScopesToConsent: this._cdfScopes });
+    location.reload();
   }
 
-  loginWithRedirect(): Promise<void> {
-    return new Promise<void>(() => {
-      const url = generateLoginUrl({ project: this.project,
-                                     baseUrl: `https://${this.cluster}.cognitedata.com`,
-                                     redirectUrl: window.location.href });
+  private setupMsalInstance(){
+    const msalConfig: Configuration = {
+      auth: {
+        clientId: this._clientId,
+        authority: `https://login.microsoftonline.com/${this._tenantId}`,
+        redirectUri: this.getRedirectUrl()
+      },
+      cache: {
+        cacheLocation: BrowserCacheLocation.LocalStorage,
+        storeAuthStateInCookie: false 
+      }
+    };
+    
+    const msalInstance = new PublicClientApplication(msalConfig);
+    const allAccounts = msalInstance.getAllAccounts();
 
-      window.location.assign(url);
-    });
+    if(allAccounts.length > 0){
+      msalInstance.setActiveAccount(allAccounts[0]);
+      this._isLoggedIn = true;
+    }
+
+    return msalInstance;
+  }
+
+  private getRedirectUrl(): string {
+    const subdomains = window.location.pathname.split('/').filter(p => p.length > 0);
+    const primarySubDomain = subdomains[0];
+    if(primarySubDomain === 'reveal-docs'){
+      return `${window.location.origin}/reveal-docs`;
+    } else if(primarySubDomain === 'reveal-docs-preview'){
+      return `${window.location.origin}/reveal-docs-preview/`;
+    }
+    throw new Error(`Unknown redirect URL: ${window.location.href}`);
   }
```

Comment:
```suggestion
  private getRedirectUrl(): string {
    const subdomains = window.location.pathname.split('/').filter(p => p.length > 0);
    const primarySubDomain = subdomains[0];
    if (primarySubDomain === 'reveal-docs') {
      return `${window.location.origin}/reveal-docs`;
    } else if (primarySubDomain === 'reveal-docs-preview') {
      return `${window.location.origin}/reveal-docs-preview/`;
    }
    throw new Error(`Unknown redirect URL: ${window.location.href}`);
  }
```
_The automatic linter has bogged down your senses, sensei_